integrations/nifi/nifi-nar-bundles/nifi-spark-bundle/nifi-spark-processors/src/main/java/com/thinkbiganalytics/nifi/v2/spark/ExecuteSparkContextJob.java

```diff
@@ -0,0 +1,256 @@
+        .expressionLanguageSupported(true)
+        .defaultValue("1")
+        .build();
+
+    public static final PropertyDescriptor NUM_CPU_CORES = new PropertyDescriptor.Builder()
+        .name("Number of CPU Cores")
+        .description("CPU Cores allocated to each executor in Spark Context")
+        .required(false)
+        .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+        .expressionLanguageSupported(true)
+        .defaultValue("2")
+        .build();
+
+    public static final PropertyDescriptor MEM_PER_NODE = new PropertyDescriptor.Builder()
+        .name("Memory Per Node")
+        .description("Memory allocated to each executor in Spark Context")
+        .required(false)
+        .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+        .expressionLanguageSupported(true)
+        .defaultValue("512mb")
+        .build();
+
+    public static final PropertyDescriptor CONTEXT_TIMEOUT = new PropertyDescriptor.Builder()
+        .name("Context Timeout")
+        .description("Automatically deletes the context after number of seconds of inactivity")
+        .required(true)
+        .addValidator(StandardValidators.INTEGER_VALIDATOR)
+        .expressionLanguageSupported(true)
+        .defaultValue("60")
+        .build();
+
+    public static final PropertyDescriptor ASYNC = new PropertyDescriptor.Builder()
+        .name("Async")
+        .description("Runs a Spark Job either asynchronously or synchronously")
+        .required(true)
+        .allowableValues("true", "false")
+        .defaultValue("false")
+        .addValidator(StandardValidators.BOOLEAN_VALIDATOR)
+        .build();
+
+    private final Set<Relationship> relationships;
+
+    private List<PropertyDescriptor> propDescriptors;
+
+    public ExecuteSparkContextJob() {
+        final Set<Relationship> r = new HashSet<>();
+        r.add(REL_SUCCESS);
+        r.add(REL_FAILURE);
+        relationships = Collections.unmodifiableSet(r);
+    }
+
+    @Override
+    protected void init(@Nonnull final ProcessorInitializationContext context) {
+        super.init(context);
+
+        // Create list of properties
+        final List<PropertyDescriptor> pds = new ArrayList<>();
+        pds.add(APP_NAME);
+        pds.add(CLASS_PATH);
+        pds.add(CONTEXT_TYPE);
+        pds.add(CONTEXT_NAME);
+        pds.add(ARGS);
+        pds.add(JOB_SERVICE);
+        pds.add(CONTEXT_TIMEOUT);
+        pds.add(ASYNC);
+        pds.add(NUM_EXECUTORS);
+        pds.add(MEM_PER_NODE);
+        pds.add(NUM_CPU_CORES);
+
+        propDescriptors = Collections.unmodifiableList(pds);
+    }
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return relationships;
+    }
+
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return propDescriptors;
+    }
+
+    @Override
+    public void onTrigger(ProcessContext context, ProcessSession session) throws ProcessException {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'org.apache.nifi.processor.exception.ProcessException' which is a runtime exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)